### ISSUE TITLE
Refactor "SwingMenuTestUtil" concerning dialog input verification methods

### ISSUE THREAD

**Issue by Husterknupp** [FRosner/DataGenerator#220]
The `SwingMenuTestUtil` has two methods doing pretty similar things while the pointing out the difference between them is quite hard. They have to be reworked or at least should the names be improved, in order to make the difference between them more obvious.
- `clickAddButtonAndCheckComponentVerification()`
- `clickEditButtonAndCheckComponentVerification()`

Same applies for:
- `openAndCancelEditFeatureDialog`
- `openAndCancelAddFeatureDialog`

**Comment by FRosner** [comment]
When thinking about it I guess that it would be possible to merge them to one method:

```
assertWorkingInputVerificationOfFeatureDefinitionDialogOpenedBy(AbstractButton button)
assertWorkingInputVerificationOfFeatureDefinitionDialogOpenedWith(AbstractButton button)
assertWorkingInputVerificationOfFeatureDefinitionDialogByClicking(AbstractButton button)
...
```

What I like about this solution:
- "assert" indicates that we have test assertions inside that method; "check" sounds like checking arguments
- clearly states what the method does and it is secondary how (using which button)

What I don't like / what we need to be aware of:
- opening the edit dialog overwrites the input fields and thus the `clickEditButtonAndCheckComponentVerification()` has some different logic; we therefore need either an if statement in the merged method or just set the input values in the add feature dialog as well (does not hurt anyone)

**Comment by FRosner** [comment]
Why is this tagged as backend? I'd rather say that input verification is a front-end issue.

Why is this tagges as duplicate? If an issue is duplicate, please clearly state which issue it is duplicating in the description.

**Comment by Husterknupp** [comment]
Oh sorry for that. I misunderstood the meaning of the duplicate tag.
- backend tag was set falsely
- duplcate tag was set because this issue concerns duplicate code and I thought the tag would also address duplicate code.
  I fixed the tagging.

**Comment by FRosner** [comment]
Yeah you cannot delete issues. So if you create one that will not be done, set it on "won't fix" and if you create one duplicate by mistake, set it "duplicate".
